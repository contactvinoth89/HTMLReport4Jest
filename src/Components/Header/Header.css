.App-header {
    background-color: #111;
    height: 50px;
    vertical-align: middle;
    text-align: center;
    font-size: calc(10px + 2vmin);
    color: white;
    top: 0;
    left: 0;
    position: fixed;
    width: 100%;
    z-index: 2;
}

input#menu {
    display: none;
}

.icon {
    cursor: pointer;
    display: block;
    height: 24px;
    padding: 16px;
    width: 24px;
}

label.icon {
    position: fixed;
    z-index: 200;
}

.icon .menu,
.icon .menu::before,
.icon .menu::after {
    background: #9fb1bd;
    content: '';
    display: block;
    height: 2px;
    position: absolute;
    transition: background ease 0.3s, top ease 0.3s 0.3s, transform ease 0.3s;
    width: 20px;
}

.icon:hover .menu,
.icon:hover .menu::before,
.icon:hover .menu::after {
    background: #47b74b;
}

.icon .menu {
    left: 18px;
    top: 27px;
}

.icon .menu::before {
    top: -6px;
}

.icon .menu::after {
    top: 6px;
}

#menu:checked + .icon .menu {
    background: transparent;
}

#menu:checked + .icon .menu::before {
    transform: rotate(45deg);
}

#menu:checked + .icon .menu::after {
    transform: rotate(-45deg);
}

#menu:checked + .icon .menu::before,
#menu:checked + .icon .menu::after {
    top: 0;
    transition: top ease 0.3s, transform ease 0.3s 0.3s;
}

.App-header span {
    line-height: 50px;
}
