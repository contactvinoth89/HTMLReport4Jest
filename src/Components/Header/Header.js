import React from 'react';
import './Header.css';
import PropTypes from 'prop-types'
export const Header = ({ menuStateChange, heading }) =>
    <div className="App-header">
        <a className="icon" onClick={() => menuStateChange('open')}>&#9776;</a>
        <span>{heading ?? 'Execution Report'}</span>
    </div>
Header.propTypes = {
    menuStateChange: PropTypes.func.isRequired,
    heading: PropTypes.string.isRequired
};
export default Header;
