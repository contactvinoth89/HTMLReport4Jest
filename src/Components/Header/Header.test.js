import '@testing-library/jest-dom'
import React from 'react';
import Header from './Header';
import { render, fireEvent } from '@testing-library/react';

describe('Header title test', () => {
    test('Should contain text sent', () => {
        const { container } = render(
            <Header heading="Test Heading" menuStateChange={function () { }}>
            </Header>
        );
        expect(container.firstChild.lastChild.textContent).toEqual("Test Heading");
    });

    test('Should contain default text if set to undeined', () => {
        const { container } = render(
            <Header heading={undefined} menuStateChange={function () { }}>
            </Header>
        );
        expect(container.firstChild.lastChild.textContent).toEqual("Execution Report");
    });

    test('Should contain default text if set to null', () => {
        const { container } = render(
            <Header heading={null} menuStateChange={function () { }}>
            </Header>
        );
        expect(container.firstChild.lastChild.textContent).toEqual("Execution Report");
    });
})

test('Should match snapshot', () => {
    const { container } = render(
        <Header heading="Test Heading" menuStateChange={function () { }}>
        </Header>
    );
    expect(container.firstChild).toMatchSnapshot();
});

test('Should fire event on click', () => {
    const FakeFun = jest.fn();
    const { container } = render(
        <Header heading="Test Heading" menuStateChange={FakeFun}>
        </Header>
    );
    fireEvent.click(container.firstChild.firstChild)
    expect(FakeFun).toHaveBeenCalled();
});