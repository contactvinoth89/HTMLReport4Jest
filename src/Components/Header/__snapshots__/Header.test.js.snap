// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header menu test Should hide menu should match snapshot 1`] = `
<div>
  <div
    class="App-header"
  >
    <span>
      Execution Report
    </span>
  </div>
</div>
`;

exports[`Should match snapshot 1`] = `
<div
  class="App-header"
>
  <input
    id="menu"
    type="checkbox"
  />
  <label
    class="icon"
    for="menu"
  >
    <div
      class="menu"
    />
  </label>
  <span>
    Test Heading
  </span>
</div>
`;
